mtools/mloginfo/sections/connection_section.py

```diff
@@ -51,6 +71,18 @@ def run(self):
                     ip, _ = tokens[3].split(':')
                 ip_opened[ip] += 1
 
+                if genstats:
+                    dt = logevent.datetime
+
+                    #initialize using default constructor
+                    ipwise_min_connection_duration[ip]
+                    ipwise_max_connection_duration[ip]
+
+                    connid = tokens[4].strip('#')
```

Comment:
Should perhaps move this into an else clause for the date comparison to avoid doing unnecessary work

Reply:
I would also consider moving the parsing of connection ID to logevent since this could be useful elsewhere.

Reply:
Agreed parsing of connection ID should be in logevent. The other code now removed as redundant with new changes.